catwalk/models/huggingface.py

```diff
@@ -52,13 +52,18 @@ def _predict_qa(
         tokenizer = cached_transformers.get_tokenizer(AutoTokenizer, self.pretrained_model_name_or_path)
         pipe = QuestionAnsweringPipeline(model=model, tokenizer=tokenizer, device=device.index or torch.cuda.current_device() if device.type == "cuda" else -1)
         
-        contexts = [instance.context for instance in converted_instances]
-        questions = [instance.question for instance in converted_instances]
-        
-        for instance, prediction in zip(converted_instances, Tqdm.tqdm(pipe(context=contexts, question=questions, batch_size=batch_size), desc="Processing instances")):
-            yield {
-                "squad_metrics": ({"id": instance.id, "prediction_text": prediction["answer"]}, {"id": instance.id, "answers": instance.answers})
-            }
```

Comment:
These changes appear to have broken batch processing. I tried adding a `batch_size=batch_size` argument to the call on `pipe` but this still produces the warning you'll see below and similarly slow compute time.

Is the progress bar issue you are attempting to fix this thing where the "Processing instances" bar finishes with one iteration to complete instead of finishing with all the iterations counted? I'm not sure why that is happening but a simpler fix using the code in `main` might be to just add `initial=1` to the `Tqdm.tqdm()` args.

Output from branch `mrqa`
```
$ python -m catwalk --model distilbert-base-cased-distilled-squad --task squad --split validation 
[08/07/22 15:02:42] WARNING  Reusing dataset squad                                                                                                builder.py:641
                             (/home/ianm/.cache/huggingface/datasets/squad/plain_text/1.0.0/d6ec3ceb99ca480ce37cdd35555d6cb2511d223b9150cce08a837               
                             ef62ffea453)                                                                                                                       
Processing instances:   3%|3         | 320/10570 [00:03<01:35, 107.70it/s]
[08/07/22 15:02:54] WARNING  /home/ianm/miniconda3/envs/catwalk_clean/lib/python3.9/site-packages/transformers/pipelines/base.py:1036:           warnings.py:109
                             UserWarning: You seem to be using the pipelines sequentially on GPU. In order to maximize efficiency please use a                  
                             dataset                                                                                                                            
                               warnings.warn(                                                                                                                   
                                                                                                                                                                
Processing instances: 100%|##########| 10570/10570 [02:25<00:00, 72.64it/s]
Calculating metrics: 100%|##########| 10570/10570 [00:12<00:00, 814.45it/s]
squad   squad_metrics.exact_match       79.57426452636719
squad   squad_metrics.f1        86.8482894897461
```

Output from branch `main`

```
$ python -m catwalk --model distilbert-base-cased-distilled-squad --task squad --split validation 
[08/07/22 14:53:36] WARNING  Reusing dataset squad                                                                                                builder.py:641
                             (/home/ianm/.cache/huggingface/datasets/squad/plain_text/1.0.0/d6ec3ceb99ca480ce37cdd35555d6cb2511d223b9150cce08a837               
                             ef62ffea453)                                                                                                                       
Processing instances: 100%|#########9| 10569/10570 [00:27<00:00, 382.56it/s]
Calculating metrics: 100%|##########| 10570/10570 [00:14<00:00, 746.27it/s]
squad   squad_metrics.exact_match       79.57426452636719
squad   squad_metrics.f1        86.8482894897461
```

Reply:
The issue I saw was that the progress bar jumped from 0 to 100% in one iteration. I did not see that error message when testing but that difference in performance is significant though. I think wrapping the data in a dataset object is one solution but isn’t very clean. I will take a closer look at this tho.

Reply:
Thanks for looking into this! Yah I guess we'll just have to see if the bug pops up again, but at least its just a UI issue.